src/metabase/server/middleware/misc.clj

```diff
@@ -79,13 +79,11 @@
 
 ;;; -------------------------------------------------- Bind request --------------------------------------------------
 
-(def ^:dynamic *request*
-  "The Ring request currently being handled by this thread, if any."
-  nil)
-
 (defn bind-request
   "Ring middleware that binds `*request*` for the duration of this Ring request."
   [handler]
   (fn [request respond raise]
-    (binding [*request* request]
-      (handler request respond raise))))
+    (request/do-with-current-request
+     request
+     (^:once fn* []
+       (handler request respond raise)))))
```

Comment:
This seems like it deserves a basic macro to wrap up the body in the `(^:once fn* [] ...)`. It seems like the usual `do-with-foo` function + `with-foo` macro pattern is in play, but skipping the macro part.

Reply:
Yeah I probably should, I just didn't want to deal with the Kondo config nonsense honestly. Laziness is not really a good excuse tho. I will fix

Reply:
fixed